src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -258,6 +258,12 @@ private DeclaredTypeAssignment GetModuleType(ModuleDeclarationSyntax syntax)
                 : new DeclaredTypeAssignment(typeRef, syntax, flags);
 
             var parent = this.binder.GetParent(syntax);
+
+            if (parent is IfExpressionSyntax ifExpressionSyntax)
+            {
+                parent = this.binder.GetParent(ifExpressionSyntax);
```

Comment:
Won't the end result here be that `ifExpressionSyntax` has no declared type? Right now, we guarantee that if you are on a node that has a declared type, you can walk up the parent pointers and everything up to the declaration will have a declared type also.

Reply:
Yeah, I missed that. This is fixed with the new grammar that Anthony suggested.